src/main/java/science/icebreaker/account/AccountService.java

```diff
@@ -132,7 +137,7 @@ public int createAccount(RegistrationRequest registration) throws AccountCreatio
      */
     public String login(Account account) {
         if (account.getId() != null) {
-            throw new IllegalArgumentException("The account id is not null.");
+            throw new AccountCreationException().withErrorCode(ErrorCodeEnum.ERR_ACC_002);
```

Comment:
This has nothing to do with account creation. So maybe better use a IllegalArgumentException or IllegalRequestParameterException.